tests/src/Functional/GroupUpdateTest.php

```diff
@@ -0,0 +1,190 @@
+    $this->createContentType(['type' => 'content_group']);
+    Og::groupManager()->addGroup('node', 'content_group');
+
+    // Create a role with only the 'update group' permission.
+    $content_editor_role = OgRole::create();
+    $content_editor_role
+      ->setName('content_editor')
+      ->setLabel('Content group editor')
+      ->setGroupType('node')
+      ->setGroupBundle('content_group')
+      ->grantPermission(OgAccess::UPDATE_GROUP_PERMISSION)
+      ->save();
+
+    // Create a group content owned by the group owner.
+    $values = [
+      'title' => 'My awesome content group',
+      'type' => 'content_group',
+      'uid' => $this->group_owner->id(),
+      'status' => 1,
+    ];
+    $this->content_group = Node::create($values);
+    $this->content_group->save();
+
+    // Subscribe the editor user to the groups.
+    $membership = OgMembership::create(['type' => OgMembershipInterface::TYPE_DEFAULT]);
+    $membership
+      ->setUser($this->group_editor->id())
+      ->setEntityId($this->content_group->id())
+      ->setGroupEntityType($this->content_group->getEntityTypeId())
+      ->setRoles([$content_editor_role->id()])
+      ->save();
+  }
+
+  /**
+   * Setup dummy entity_test group entity and appropriate og permissions.
+   */
+  public function setUpEntityTestEntity() {
+    // Create an entity_test bundle called 'entity_group' and make it
+    // an Og group.
+    entity_test_create_bundle('entity_group');
+    Og::groupManager()->addGroup('entity_test', 'entity_group');
+
+    // Create a role with only the 'update group' permission.
+    $entity_editor_role = OgRole::create();
+    $entity_editor_role
+      ->setName('entity_editor')
+      ->setLabel('Entity group editor')
+      ->setGroupType('entity_test')
+      ->setGroupBundle('entity_group')
+      ->grantPermission(OgAccess::UPDATE_GROUP_PERMISSION)
+      ->save();
+
+    // Create an entity_test entity owned by the group owner.
+    $values = [
+      'title' => 'My awesome group',
+      'type' => 'entity_group',
+      'uid' => $this->group_owner->id(),
+    ];
+    $this->entity_group = EntityTest::create($values);
+    $this->entity_group->save();
+
+    $membership = OgMembership::create(['type' => OgMembershipInterface::TYPE_DEFAULT]);
+    $membership
+      ->setUser($this->group_editor->id())
+      ->setEntityId($this->entity_group->id())
+      ->setGroupEntityType($this->entity_group->getEntityTypeId())
+      ->setRoles([$entity_editor_role->id()])
+      ->save();
+  }
+
+  /**
+   * Tests 'update group' special permission.
+   *
+   * @dataProvider ogUpdateAccessProvider
+   */
+  public function testUpdateAccess($entity) {
+    // The editor should have permissions due to the 'update group' special
+    // permission.
+    $this->drupalLogin($this->group_editor);
+    $this->drupalGet($this->{$entity}->toUrl('edit-form'));
+    $this->assertSession()->statusCodeEquals(200);
```

Comment:
> oh, you mean that via Kernel test you were not getting this edit? 

I think I actually remembered something wrong because I remember checking directly the roles for having the roles and not through the userAccess method but apparently I was mistaking. I will also extend the OgEntityAccess to add a test for this.
Still, my initial thought still remains. This is a special permission that needs mapping. I thought that a real case scenario where Drupal requires the permission will assure as that the operation requested is ok and the permission will be granted successfully. Is this a bat approach?